### PR TITLE
Disable alerts in vendored dirs

### PR DESCRIPTION
**Pull request by NathanReb**
Alerts weren't silenced when compiling code in vendored directories. Same as for warnings, users likely do not want to modify the code they vendor and therefore are not in a position to do anything about the alerts.

This PR extends the treatment of warnings in vendored directories to alerts, simply disabling all alerts when compiling code in such directories.

Alerts were introduced in OCaml 4.08 so the `-alert -all` is only added if the version of OCaml is greater or equal to 4.08.

### CODE REVIEW

File: src/dune_rules/super_context.ml [hunk 1]
```diff
@@ -354,6 +354,12 @@ let build_dir_is_vendored build_dir =
   | Some src_dir -> Dune_engine.Source_tree.is_vendored src_dir
   | None -> Memo.return false
 
+let add_vendored_flags ~ocaml_version flags =
```
Comment (voodoos) [inline src/dune_rules/super_context.ml:357]: Maybe calling this function `with_vendored_flags` would be more coherent ?

File: test/blackbox-tests/test-cases/vendor/main.t/run.t [hunk 2]
```diff
@@ -15,6 +15,12 @@ When compiling vendored code, all warnings should be disabled
   Entering directory 'warnings'
   There should be no OCaml warning!
 
+When compiling vendored code, all alerts should be disabled as well
+
+  $ dune build --root alerts @no-alerts-please
+  Entering directory 'alerts'
+  There should be no OCaml alerts!
+
```
Comment (voodoos) [inline test/blackbox-tests/test-cases/vendor/main.t/run.t:23]: Maybe we should also test that warnings / alerts are indeed shown if the `vendored_dirs` stanza is removed ?

**Review by voodoos — APPROVED** [review]
These changes look good to me.

File: src/dune_rules/super_context.ml (hunk 1 above)
Comment (NathanReb) [reply src/dune_rules/super_context.ml:357]: Sure!

File: test/blackbox-tests/test-cases/vendor/main.t/run.t [hunk 3]
```diff
@@ -15,6 +15,12 @@ When compiling vendored code, all warnings should be disabled
   Entering directory 'warnings'
   There should be no OCaml warning!
 
+When compiling vendored code, all alerts should be disabled as well
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/vendor/main.t/run.t:18]: Could you put this into a separate test file or directory? E.g. `vendor/alerts.t`?

Comment (NathanReb) [reply test/blackbox-tests/test-cases/vendor/main.t/run.t:18]: Sure!

**Comment by rgrinberg** [comment]
Nice work. Merged.

**Comment by emillon** [comment]
I think there was an error while merging since tests are failing now. @NathanReb can you have a look?

**Pull request closed without merging**